### ISSUE TITLE
FT991A-PAT components are unobtanium due to supply chain madness

### ISSUE THREAD

**Issue by ghost** [rfrht/FT991A-PAT#21]
Hi, thank you (and W3AXL, and OE2DOR) for designing this awesome board and sharing it with us!

I've ordered my PCBs from OSHPark, but discovered that five of the components on the BOM are not available, due to general supply chain insanity (which has been going on for a couple of years now...).

I went ahead and ordered from Digikey anyways, with the following four replacements; do you think these choices will work?  If not, can you suggest better substitutions?

* [MC78L09ABDR2GOSCT-ND](https://www.digikey.com/en/products/detail/onsemi/MC78L09ABDR2G/1481351) -> [296-11126-5-ND](https://www.digikey.com/en/products/detail/texas-instruments/UA78L09ACD/382182)
* [NCP553SQ30T1GOSCT-ND](https://www.digikey.com/en/products/detail/onsemi/NCP553SQ30T1G/920092) -> [1662-1980-1-ND](https://www.digikey.com/en/products/detail/ablic-inc/S-817A30ANB-CUTT2U/6120535)
* [MMBFJ310LT1GOSCT-ND](https://www.digikey.com/en/products/detail/onsemi/MMBFJ310LT1G/919591) -> [MMBFJ211CT-ND](https://www.digikey.com/en/products/detail/onsemi/MMBFJ211/1049196)
* [2SC5086-OLFCT-ND](https://www.digikey.com/en/products/detail/toshiba-semiconductor-and-storage/2SC5086-O-LF/4516268) -> [BFR106E6327HTSA1CT-ND](https://www.digikey.com/en/products/detail/infineon-technologies/BFR106E6327HTSA1/1281529)

The real problem, however, is [USB2514BI-AEZG-CT-ND](https://www.digikey.com/en/products/detail/microchip-technology/USB2514BI-AEZG-TR/3873166). That is total unobtanium right now, Digikey is showing an 18-month lead time.  And it isn't a commodity part.  Any ideas about what to do here?  Should I try buying an old USB hub on eBay and scavenging from it?  Any ideas which high-volume products might have used this IC?

Thanks again,

**Comment by rfrht** [comment]
Hello there, Joseph - and thanks for getting in touch!

Man, I've been for the last two hours drowning in datasheets and here's what I've found.

* The 9V regulator (MC78) is still stocked, so I didn't evaluate the substitute.
* The 3.3V regulator (NCP553) replacement (S-817) looks good. But the price, ugh...
* The J310 - I was able to find an even better mach. Replace with the [U310](https://www.digikey.com/en/products/detail/onsemi/mmbfu310lt1g/919592).
* Now, the 2SC5086... Argh. We have few problems. First, the package is proprietary. The BFR106 that you pointed looks okay in its specs, but the footprint won't match the 2SC5086 in the board (the BFR106 is a SOT-23).

I am preparing [Revision S](https://github.com/rfrht/FT991A-PAT/tree/Rev-S-Full) replacing the 2SC5086 with a [BFR193](https://www.digikey.com/en/products/detail/infineon-technologies/BFR193FH6327XTSA1/2337789), but I am not entirely convinced about that part. Anyways; I have ordered the BFR193 and will made its trip to here in some more weeks. Ah, the BFR193 is *also* not compatible with the 2SC5086 footprint.

Matter of fact, I was thinking about selecting other component; capping a good lot the Ft (frequency transition), because we are in need of a product which usage is in the **low** VHF frequency, ~70 MHz. Using so low frequencies with products meant to use in higher frequencies is not a great idea. And the bias will probably need some attention.

After these two hours fiddling with datasheets, availability, etc., I was flirting with an Infineon [BGA614](https://www.digikey.com/en/products/detail/infineon-technologies/BGA614H6327XTSA1/2337815), but man... Lots of testing would have to be done, redo the bias network etc.

So tl;dr: As of now, it is... Complicated.

The USB2514, phew... last time I bought it was in 2020. I had a spare one and resold to a friend who bought the panadapter board from me.

Ah; you missed the PE4259 (I **strongly** recommend the RF switch to take the panadapter out when transmitting, it causes some shenanigans when transmitting) - that was pretty much unobtanium too. Fortunately I do have some here, and if you found where to buy let me know, in case someone ask.

Let me know if you have any further comments. Thanks for stopping by! o/

**Comment by rfrht** [comment]
Note to self: The BFR193F was chosen because it is listed as a 2SC5086 replacement in [Infineon's RF Discretes Selection Guide](https://rf3.org:8443/q/infineon-rf.pdf)

**Comment by ghost** [comment]
> The USB2514, phew... last time I bought it was in 2020. I had a spare one and resold to a friend who bought the panadapter board from me.

Ugh, not what I was hoping to hear :(

Microchip just published a production change notice indicating that they have added a second IC packaging provider for the product, so presumably production has resumed, but that certainly hasn't been reflected in the lead times.

Looks like there are a few grey-market sources on ebay, at least one of which looks not-completely-shady, so I will roll the dice on that one.

> The 9V regulator (MC78) is still stocked, so I didn't evaluate the substitute.

Oh wow, they must've just gotten a shipment.  Sweet.

> * First, the package is proprietary

WTF?  Why, oh why, must manufacturers do things like this...

> Matter of fact, I was thinking about selecting other component; capping a good lot the Ft (frequency transition)

Thank you so much for looking at this!  My analog experience is limited to VLSI and I have zero experience with RF design on any level, so I really appreciate it.

> Ah; you missed the PE4259

You're right.  I substituted [PE42421SCAA-Z](https://www.digikey.com/en/products/detail/psemi/PE42421SCAA-Z/3728588); do you see any problems there?  I felt a bit safer with that substitution since the gate on that switch is a (very) low-frequency signal.

I'm closing this (thanks again for your response!) but will follow up with a report on how things turn out.

**Comment by rfrht** [comment]
*Reopening it and leaving it open as it contains useful information to the wider audience - added reference in Readme.*

Hello again Joseph! 

Definitely, the [PE42421](https://www.digikey.com/en/products/detail/psemi/PE42421SCAA-Z/3728588) is **not** a suitable candidate. It's exactly the same part as PE4259, **except** that the RF control flips the RF1/RF2 ports the other way. Ah and it is not stocked at DigiKey too.

I was evaluating the Skyworks [SKYA21001](https://www.skyworksinc.com/-/media/SkyWorks/Documents/Products/1901-2000/SKYA21001_202936H.pdf) or [AS213-92](https://www.skyworksinc.com/-/media/SkyWorks/Documents/Products/1-100/200193D.pdf) - but the RF switch works differently from the 4259; it is controlled by powering **either** RFC or VCC. Perfect pinout match, but the logic doesn't work.

**Comment by rfrht** [comment]
Update.

I have checked all [RF switches, in stock, SPDT, SC-70 package](https://www.digikey.com/en/products/filter/rf-switches/865?s=N4IgjCBcpgzADFUBjKAzAhgGwM4FMAaEAeygG1wAmagdhpCIBZ4BWATgDZ6BdIgBwAuUEAGUBAJwCWAOwDmIAL5EwlNhGghUkTLkIlyIZvErx6TeMfgcQvEIOFipcxUQC0lJJqgSArntKQFCw2CqFAA), and **none** are directly compatible with PE4259, the way it was implemented in the panadapter board:

The PE4259 supports two switching modes:

* Vcc always high & RFC high/low (the design choice)
* **Either** VCC or RFC high

Welp, seems that the *either* VCC or RFC mode is the standard/de facto, as **every product** supports it, with only the PE4259 allowing the VCC + RFC state mode.

This design make sense in the panadapter, as it would require another component to drive it accordingly, but... Well, makes it hard (impossible?) to find replacements.

**Comment by rfrht** [comment]
USB2514BI [Microchip.com store](https://www.microchip.com/en-us/product/USB2514B#buy-from-store): Ships on May, 2023

Other [compatible package (didn't check for pinout compatibility) USB controllers](https://www.digikey.com/en/products/filter/interface-controllers/753?s=N4IgjCBcpgbFoDGUBmBDANgZwKYBoQB7KAbXAE4B2AFjACYQBdAgBwBcoQBlNgJwEsAdgHMQAXwlA): None in stock

😐

**Comment by ghost** [comment]
> I have checked all [RF switches, in stock, SPDT, SC-70 package](https://www.digikey.com/en/products/filter/rf-switches/865?s=N4IgjCBcpgzADFUBjKAzAhgGwM4FMAaEAeygG1wAmagdhpCIBZ4BWATgDZ6BdIgBwAuUEAGUBAJwCWAOwDmIAL5EwlNhGghUkTLkIlyIZvErx6TeMfgcQvEIOFipcxUQC0lJJqgSArntKQFCw2CqFAA), and **none** are directly compatible with PE4259, the way it was implemented in the panadapter board:

I guess I should wait for the next revision of the board.

**Comment by rfrht** [comment]
Hello @amjoseph-nixpkgs 

Just pushed a commit that might address this issue. It's on [Revision S Lite](https://github.com/rfrht/FT991A-PAT/tree/Rev-S-Lite) tree

In this revision I am using the "compatibility" way to flip the PE4259, which would essentially allow to replace it by any other RF switch with compatible component footprint.

**Comment by ghost** [comment]
Boards+stencil ordered from OSHPark; will let you know when it arrives and I build it.

I did in fact manage to get my hands on 10pcs of the unobtanium USB chip from an extremely unofficial distributor.  If you want the leftovers (8-9pcs probably) I'd be happy to mail them to you (once I verify that I didn't get swindled into buying a bunch of empty QFN packages!) 

> In this revision I am using the "compatibility" way to flip the PE4259, which would essentially allow to replace it by any other RF switch with compatible component footprint.

Is there a specific replacement part you can recommend at this time?

**Comment by rfrht** [comment]
> Boards+stencil ordered from OSHPark; will let you know when it arrives and I build it.

Hooray!!

> I did in fact manage to get my hands on 10pcs of the unobtanium USB chip from an extremely unofficial distributor. If you want the leftovers (8-9pcs probably) I'd be happy to mail them to you (once I verify that I didn't get swindled into buying a bunch of empty QFN packages!)

That was... Bold! Good luck!

> Is there a specific replacement part you can recommend at this time?
Ah, supply chain. Well, there's a pSemi part with slightly lower specs (though better than Skyworks) and reasonable pricing: [PE42421SCAA-Z](https://www.digikey.com/en/products/detail/psemi/PE42421SCAA-Z/3728588).

Fun fact: The RF switch logic follows the same as PE4259 and is a *direct replacement* part. It's just the specs that are a bit (very small bit) degraded.

**Comment by rfrht** [comment]
Very well. After more than a year, I finally moved, inventoried my parts, will test the new revision board (S) and I was able to - other than the 2SC5086 EOLed part - order everything.

Closing this issue.
